contrib/ext_editor.lua

```diff
@@ -0,0 +1,371 @@
+	TODO
+	* localization
+	* buttons are not equal in size and centered
+	
+    BUGS, COMMENTS, SUGGESTIONS
+    * send to Marco Carrarini, marco.carrarini@gmail.com
+
+    CHANGES
+    * 20191220 - initial version, PR done
+	* 20191221 - reworked, cleaned, added "copy and edit"
+	* 20191222 - added darkroom mode
+	* 20191223 - check API version, OS compatibility
+	
+]]
+
+
+local dt = require "darktable"
+local du = require "lib/dtutils"
+local df = require "lib/dtutils.file"
+local dtsys = require "lib/dtutils.system"
+
+
+-- check API version
+du.check_min_api_version("5.0.2", "ext_editor")  -- darktable 3.0
+
+
+-- OS compatibility
+local PS = dt.configuration.running_os == "windows" and  "\\"  or  "/"
+
+
+-- executables of the external editors
+local program_paths = {
+	[1] = dt.preferences.read("ext_editor","program_path_1", "string"),
+	[2] = dt.preferences.read("ext_editor","program_path_2", "string"),
+	[3] = dt.preferences.read("ext_editor","program_path_3", "string"),
+	[4] = dt.preferences.read("ext_editor","program_path_4", "string"),
+	[5] = dt.preferences.read("ext_editor","program_path_5", "string"),
+	[6] = dt.preferences.read("ext_editor","program_path_6", "string"),
+	[7] = dt.preferences.read("ext_editor","program_path_7", "string"),
+	[8] = dt.preferences.read("ext_editor","program_path_8", "string"),
+	[9] = dt.preferences.read("ext_editor","program_path_9", "string")
+	}
+
+
+-- friendly names of the external editors
+local program_names = {
+	[1] = dt.preferences.read("ext_editor","program_name_1", "string"),
+	[2] = dt.preferences.read("ext_editor","program_name_2", "string"),
+	[3] = dt.preferences.read("ext_editor","program_name_3", "string"),
+	[4] = dt.preferences.read("ext_editor","program_name_4", "string"),
+	[5] = dt.preferences.read("ext_editor","program_name_5", "string"),
+	[6] = dt.preferences.read("ext_editor","program_name_6", "string"),
+	[7] = dt.preferences.read("ext_editor","program_name_7", "string"),
+	[8] = dt.preferences.read("ext_editor","program_name_8", "string"),
+	[9] = dt.preferences.read("ext_editor","program_name_9", "string")
+	}
+
+
+-- last used editor
+if not (dt.preferences.read("ext_editor","initialized", "bool")) then
+	dt.preferences.write("ext_editor","lastchoice", "integer", 1)
+	dt.preferences.write("ext_editor","initialized", "bool", true)
+	end
+local lastchoice = dt.preferences.read("ext_editor","lastchoice", "integer")
+
+
+-- shows export progress
+local function show_status(storage, image, format, filename,
+	number, total, high_quality, extra_data)
+    dt.print(string.format("Exporting Image %i/%i ...", number, total))
+	end
+
+
+-- callback for button "edit"
+local function OpenWith(images, choice)
+		
+	if #images == 1 then
+		for _, image in pairs(images) do 
+			
+			-- image to be edited
+			name = image.path..PS..image.filename 					
+			
+			-- launch the external editor
+			local run_cmd = df.sanitize_filename(program_paths[choice]).." "..df.sanitize_filename(name) 
```

Comment:
You should check and see if program_paths[choice] is actually and executable or not, just in case someone made a typo when they typed in the path, or a file got moved, etc.  df.check_if_bin_exists() returns the sanitized executable command (i.e. prefixes wine when trying to run a .exe file on linux or macos).